File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.java
class Solution {
    static int result = 0;
    public int solution(int a, int b, int n) {
        if (n < a) {
            return result;
        }
        int re = 0;
        while (n >= a) {
            re += b;
            n -= a;
        }
        n += re;
        result += re;
        return solution(a, b, n);
    }
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.swift
import Foundation

func solution(_ a:Int, _ b:Int, _ n:Int) -> Int {
    var cokeCount = n
    var receivedCount = 0
    
    while cokeCount >= a {
        receivedCount += (cokeCount / a * b)
        cokeCount = (cokeCount / a * b) + (cokeCount % a)
    }
    
    return receivedCount
}